test.js

```diff
@@ -91,3 +92,40 @@ test('kill from port', async t => {
 test('error when process is not found', async t => {
 	await t.throwsAsync(fkill(['notFoundProcess']), /Killing process notFoundProcess failed: Process doesn't exist/);
 });
+
+function testKillDescendant(t, name = '') {
+	const fixture = path.resolve('fixtures', 'descendant');
+	const cp = childProcess.spawn('node', [fixture, name]);
+
+	cp.stdout.setEncoding('utf8');
+	cp.stdout.on('data', async chunk => {
+		const descendantPid = parseInt(chunk, 10);
+		t.is(typeof descendantPid, 'number');
+
+		const opts = {
+			tree: true,
+			force: process.platform === 'win32'
+		};
+
+		if (name) {
+			await fkill(name, opts);
+		} else {
+			await fkill(cp.pid, opts);
+		}
+
+		// Ensure all the processes has time to exit
+		await delay(400);
+
+		t.false(await processExists(cp.pid));
+		t.false(await processExists(descendantPid));
+		t.end();
+	});
+}
+
+// eslint-disable-next-line ava/test-ended
+test.cb('kill all descendants tree by pid', testKillDescendant);
+
+if (process.platform !== 'win32') {
+	// eslint-disable-next-line ava/test-ended
+	test.cb('kill all descendants tree by name', testKillDescendant, 'fkill-descen');
```

Comment:
```suggestion
	test.cb('kill all descendants tree by name', testKillDescendant, 'fkill-descendant');
```

?

Reply:
Sorry for not mentioning it before.
I've used the string "fkill-descen" because it's 12 chars, which is the theoretical maximum of chars I can change on `process.title` because of the limitation that have Linux and macOS as explained in the [Node.js `process.title` docs](https://nodejs.org/api/process.html#process_process_title).
If I'd got above the limit it might not match the process title and fail the test.